blazar/plugins/devices/k3s_plugin.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+#
+# Author: Pierre Riteau <pierre@stackhpc.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+from oslo_config import cfg
+
+from blazar.db import api as db_api
+from blazar.db import exceptions as db_ex
+from blazar.manager import exceptions as manager_ex
+from kubernetes import client
+from kubernetes import config
+from oslo_log import log as logging
+
+opts = [
+    cfg.StrOpt(
+        'kubeconfig_file', help='Kubeconfig file to use for calls to k8s'),
+]
+
+CONF = cfg.CONF
+CONF.register_opts(opts, group="k8s")
+LOG = logging.getLogger(__name__)
+
+LABEL_NAMESPACE = "blazar.openstack.org"
+LABELS = {
+    "reservation_id": f"{LABEL_NAMESPACE}/reservation_id",
+    "device": f"{LABEL_NAMESPACE}/device",
+}
+
+
+class K3sPlugin():
+    device_driver = 'k3s'
+
+    def __init__(self):
+        config.load_kube_config(config_file=CONF.k8s.kubeconfig_file)
+        self.core_v1 = client.CoreV1Api()
+        self.apps_v1 = client.AppsV1Api()
+
+    def has_label(self, node, label, value):
+        '''Get if the node has label=value, or if value is none, any value'''
+        return (label in node.metadata.labels and
+                (value is None or node.metadata.labels.get(label) == value))
+
+    def set_label(self, name, label, value):
+        body = {
+            "metadata": {
+                "labels": {
+                    label: value,
+                },
+            },
+        }
+        return self.core_v1.patch_node(name, body)
+
+    def set_res_id_label(self, name, reservation_id):
+        return self.set_label(
+            name, LABELS["reservation_id"], reservation_id)
+
+    def set_device(self, name, value):
+        return self.set_label(name, LABELS["device"], value)
+
+    def get_node_by_name(self, name):
+        node_list = self.core_v1.list_node()
```

Comment:
There exists `read_node(name)`, which I think it more straightforward.

Reply:
Thank you for the suggestion!